src/ImageSharp/Formats/Jpeg/JpegDecoderCore.cs

```diff
@@ -1022,10 +1022,26 @@ private void ProcessStartOfFrameMarker(BufferedReadStream stream, int remaining,
             int index = 0;
             for (int i = 0; i < componentCount; i++)
             {
+                // 1 byte: component identifier
+                byte componentId = this.temp[index];
+
+                // 1 byte: component sampling factors
                 byte hv = this.temp[index + 1];
                 int h = (hv >> 4) & 15;
                 int v = hv & 15;
 
+                // Validate: 1-4 range
+                if (h is < 1 or > 4)
+                {
+                    JpegThrowHelper.ThrowBadSampling(h);
+                }
+
+                // Validate: 1-4 range
+                if (v is < 1 or > 4)
```

Comment:
This reads really nice, but codegen has to two conditional jumps. With uint and subtraction, there could be only I conditional jump.

Therefore I propose a helper like
```c#
static bool IsOutOfRange(int value, int min, int max)
    => (uint)(value - min) > (uint)(max - min);
```
then it can be written as
```suggestion
                if (IsOutOfRange(v, 1, 4))
```
The JIT will use the constants, and produce special inlined code for that.

For codegen see [example on sharplab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKBuIGYACMxgYUYG8bGfmnjySRsAgQANowCy5ABQBLAHYZGANwCUjALwA+RjJWM5uRgB5G5RtEa6UagNzdeDZoOGiJk0vKWqNOxgCSuADyAK4YwQBmAErYCgDmMPpo5im2DtS8jI48zgJCisoAYqKyhb45nJVZcpF6Bkam5pZQ1oy21bxcmVm9PAAqABZQEADuMvadPAC+U9k9fcQA7KoZWbML81l5ruUlEF7l6pXdfTy1ekFhETFxicmp7Wpqc6dnWUMj45Ob63Nzy1WlQ2vEqlQADlA5CpsBgYC4hCJxIEQuEorEEkkjtgxKEYClyvhFASfPhsAgXr9eP4ZDJQoU1PocXjGHBGESFM82rT6UpGWSEKz2YpnmtQZtIdDYfD8swUIxPmMJlpdBhhmNGAoYKNGABRBBgGDgjByCAKCYZaZAA==)

Reply:
Note that this is not a hot path, but if we create a reusable `Numerics.IsOutOfRange` method, I'm not against. Up to @br3aker to decide.

Reply:
I'm up for a helper method. I personally do and do not like pattern matching at the same time. And if it has worse performance - let's introduce `IsOutOfRange` method.

Reply:
>  I personally do and do not like pattern matching at the same time.

Same here. It fragments the language, pushing it towards the C++ direction where there are literally 5-10 different ways to achieve the same thing sometimes.

Reply:
Done, I've also added tests for new method.